pkg/util/utils.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2022 Antrea Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strings"
+
+	"github.com/ClickHouse/clickhouse-go"
+	"github.com/google/uuid"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+var SqlOpenFunc = sql.Open
+
+func ParseRecommendationName(npName string) error {
+	if !strings.HasPrefix(npName, "pr-") {
+		return fmt.Errorf("input name %s is not a valid policy recommendation job name", npName)
+
+	}
+	id := npName[3:]
+	_, err := uuid.Parse(id)
+	if err != nil {
+		return fmt.Errorf("input name %s does not contain a valid UUID, parsing error: %v", npName, err)
+	}
+	return nil
+}
+
+func GetServiceAddr(client kubernetes.Interface, serviceName, serviceNamespace, servicePortName string) (string, int, error) {
+	var serviceIP string
+	var servicePort int
+	service, err := client.CoreV1().Services(serviceNamespace).Get(context.TODO(), serviceName, metav1.GetOptions{})
+	if err != nil {
+		return serviceIP, servicePort, fmt.Errorf("error when finding the Service %s: %v", serviceName, err)
+	}
+	serviceIP = service.Spec.ClusterIP
+	for _, port := range service.Spec.Ports {
+		if port.Name == servicePortName {
+			servicePort = int(port.Port)
+		}
```

Comment:
Maybe we can check `port.Protocol` as we do previously in CLI util instead of `port.Name` to avoid the naming problem?

Reply:
Sure. Updated.